homeassistant/components/reolink/__init__.py

```diff
@@ -138,6 +149,7 @@ async def async_check_firmware_update() -> str | Literal[False]:
         config_entry.add_update_listener(entry_update_listener)
     )
 
+    starting = False
```

Comment:
With the `async_refresh` above, even though it's awaited, does it guarantee that the first run-through of the firmware update has completed by the time the code hits this?
I thought `async_refresh` was deferred with a debounce timer, in which case the check for `starting` above may always see `False`?

Reply:
Yes, `async_refresh` guarantees that it is completed.
`async_refresh` is not deffered, `async_request_refresh` is

see: https://github.com/home-assistant/core/blob/4860a8d1e8853609c13fc4e7d0f962f9804276cf/homeassistant/helpers/update_coordinator.py#L235-L258